File: project-euler.go/solutions/_8/8.go
package _8

import "project-euler/common/data"

var digits []int

func Initialize() {
	for _,b := range data.ReadNoNewLines() {
		digits = append(digits, int(b - 48))
	}
}

func Solve() int64 {
	var max int64
	for i := 0; i < len(digits) - 12; i++ {
		var p int64 = 1
		for j := 0; j < 13; j++ {
			p *= int64(digits[i + j])
		}

		if (p > max) {
			max = p;
		}
	}

	return max;
}

File: project-euler.java/src/main/java/euler/solutions/_8.java
package euler.solutions;

import euler.common.BaseSolution;
import gnu.trove.list.TIntList;
import gnu.trove.list.array.TIntArrayList;

public class _8 extends BaseSolution {
    private int[] digits;

    @Override
    public void initialize() throws Exception {
        TIntList digitsList = new TIntArrayList();
        for (char chr : dataFileStringNoNewlines().toCharArray()) {
            digitsList.add((int)chr - 48);
        }
        digits = digitsList.toArray();
    }

    @Override
    public long solve() {
        long max = 0;
        for (int i = 0; i < digits.length - 12; i++) {
            long p = 1;
            for (int j = 0; j < 13; j++) {
                p *= digits[i + j];
            }

            if (p > max) {
                max = p;
            }
        }

        return max;
    }
}
